src/grt/src/RepairAntennas.cpp

```diff
@@ -855,4 +857,623 @@ double RepairAntennas::diffArea(odb::dbMTerm* mterm)
+  int last_via_y = pos_y;
+  // iterate all position of segment
+  while (validatePos(is_reversed, pos_x, pos_y, seg_final_x, seg_final_y)) {
+    // check if there are vias in the position
+    if (vias_pos.find(std::make_pair(pos_x, pos_y)) != vias_pos.end()
+        || !grouter_->hasCapacity(
+            is_horizontal, pos_x, pos_y, seg.init_layer + 2)) {
+      // get size from last vias to the position
+      const int free_via_size
+          = std::abs(pos_x - last_via_x) + std::abs(pos_y - last_via_y);
+      // check if the jumper can be added in the sub segment
+      if (free_via_size > 0 && free_via_size >= bridge_size + 2 * tile_size) {
+        // calculate the size from init position to add jumper
+        req_space_to_add = std::abs(last_via_x - seg_init_x)
+                           + std::abs(last_via_y - seg_init_y) + tile_size;
+      }
+      last_via_x = pos_x;
+      last_via_y = pos_y;
+    }
+    if (is_horizontal) {
+      pos_x += step;
+    } else {
+      pos_y += step;
+    }
+  }
+  // if the segment has no vias in the end, verify last sub segment
+  if (last_via_x != seg_final_x || last_via_y != seg_final_y) {
+    const int free_via_size = std::abs(seg_final_x - last_via_x)
+                              + std::abs(seg_final_y - last_via_y);
+    if (free_via_size > 0 && free_via_size >= bridge_size + 2 * tile_size) {
+      req_space_to_add = std::abs(last_via_x - seg_init_x)
+                         + std::abs(last_via_y - seg_init_y) + tile_size;
+    }
+  }
+  return req_space_to_add;
+}
+
+int RepairAntennas::getSegmentIdToAdd(std::vector<int>& segment_ids,
+                                      const GRoute& route,
+                                      int& req_size,
+                                      const int& bridge_size,
+                                      const int& tile_size,
+                                      bool is_horizontal,
+                                      bool in_start)
+{
+  // Sort segments
+  if (is_horizontal) {
+    sort(segment_ids.begin(),
+         segment_ids.end(),
+         [&route](const int& seg_id1, const int& seg_id2) {
+           return route[seg_id1].init_x < route[seg_id2].init_x;
+         });
+  } else {
+    sort(segment_ids.begin(),
+         segment_ids.end(),
+         [&route](const int& seg_id1, const int& seg_id2) {
+           return route[seg_id1].init_y < route[seg_id2].init_y;
+         });
+  }
+  const int layer_level = route[segment_ids[0]].init_layer;
+  const auto& vias_pos = vias_pos_[layer_level];
+  int size_sum = 0;
+  int best_seg_id = -1;
+  int best_req_space = 0;
+  bool is_reversed = false;
+  // Reverse the segments when jumper is added near the end (max)
+  if (!in_start) {
+    is_reversed = true;
+    std::reverse(segment_ids.begin(), segment_ids.end());
+  }
+  for (int pos = 0; pos < segment_ids.size(); pos++) {
+    const auto& seg = route[segment_ids[pos]];
+    // if seg could have jumper, get via position
```

Comment:
This comment seem out of place. I can't find in the code where you're getting the via position.